preprocessors/openstreetmap/osm_service.py

```diff
@@ -398,10 +432,17 @@ def enlist_POIs(processed_OSM_data1, amenity):
                         # Check to remove duplicate intersections
                         if nodes[node] not in POIs:
                             POIs.append(nodes[node])
+                            if nodes[node]["id"] not in nodes_ids:
+                                nodes_ids.append(nodes[node]["id"])
+                                unique_POIs.append(nodes[node])
+
     if amenity is not None and len(amenity) != 0:
         for objs in range(len(amenity)):
             POIs.append(amenity[objs])
-    return POIs  # POIs is a list of all points of interest
+            unique_POIs.append(amenity[objs])
+            # POIs is a list of all points of interest
+            # unique_POIs is an unduplicated list of all points of interest
+    return POIs, unique_POIs
```

Comment:
I don't see why it's necessary to return both a list with duplicates and a deduplicated list. Could you please explain this?

Reply:
I could observe some level of redundancy in the code actually.  I was only initially trying to avoid the possibility of making multiple alterations to the code, but I've also now realized that it doesn't have to be. So, I've corrected it now.